libraries/standard/mqtt/src/mqtt.c

```diff
@@ -214,6 +233,122 @@ static MQTTStatus_t sendPublish( const MQTTContext_t * const pContext,
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Calculate the interval between two timestamps, including when the
+ * later value has overflowed.
+ *
+ * @param[in] later The later time stamp.
+ * @param[in] start The earlier time stamp.
+ *
+ * @return later - start.
+ */
+static uint32_t calculateElapsedTime( uint32_t later,
+                                      uint32_t start )
+{
+    return later - start;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t receivePacket( MQTTContext_t * const pContext,
+                                   MQTTPacketInfo_t incomingPacket,
+                                   uint32_t remainingTimeMs )
+{
+    return MQTTSuccess;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
+                                    uint32_t timeoutMs,
+                                    MQTTPacketInfo_t * const incomingPacket,
+                                    bool * const pSessionPresent )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    MQTTGetCurrentTimeFunc_t getTimeStamp = NULL;
+    uint32_t entryTime = 0U, remainingTimeMs = timeoutMs;
+
+    assert( pContext != NULL );
+    assert( incomingPacket != NULL );
+
+    getTimeStamp = pContext->callbacks.getTime;
+    /* Get the entry time for the function. */
+    entryTime = getTimeStamp();
+
+    do
+    {
+        /* Transport read for incoming CONNACK packet type and length. */
+        status = MQTT_GetIncomingPacketTypeAndLength( pContext->transportInterface.recv,
+                                                      pContext->transportInterface.networkContext,
+                                                      incomingPacket );
+
+        /* Loop until there is data to read or if the timeout has not expired. */
+    } while( ( status == MQTTNoDataAvailable ) &&
+             ( calculateElapsedTime( getTimeStamp(), entryTime ) < timeoutMs ) );
+
+    if( status == MQTTSuccess )
+    {
+        /* Check if the timeout has expired. */
```

Comment:
I think we should make at least one read attempt if we have successfully received type and length. Basically time check block can be deleted.

Reply:
Yes Done